src/foam/nanos/dig/DigWebAgent.java

```diff
@@ -339,11 +342,64 @@ public void execute(X x) {
         }
         out.println(returnMessage);
       } else if ( Command.select == command ) {
+        resp.setContentType("text/html");
+
         PropertyInfo idProp = (PropertyInfo) cInfo.getAxiomByName("id");
         ArraySink sink = (ArraySink) ( ! SafetyUtil.isEmpty(id) ?
           dao.where(MLang.EQ(idProp, id)).select(new ArraySink()) :
           dao.select(new ArraySink()));
 
+        SinkParser sinkParser = new SinkParser(cInfo);
+        StringPStream ps = new StringPStream(fields);
+        ParserContextImpl x_ = new ParserContextImpl();
+        ps = (StringPStream) sinkParser.parse(ps, x_);
+
+        Object[] psArray = (Object[]) ps.value();
+        logger.debug("ps.value.length: " + psArray.length);
+
+        for ( Object objPs : psArray ) {
+          logger.debug("ps.value() : " + objPs);
+
+          if ( objPs.toString().startsWith("groupBy") || objPs.toString().startsWith("min") || objPs.toString().startsWith("max") || objPs.toString().startsWith("avg") ) {
+            String pName = getPropertyName(objPs.toString());
+            PropertyInfo pInfo = (PropertyInfo) cInfo.getAxiomByName(pName);
+
+            if ( pInfo != null) {
+              if ( objPs.toString().startsWith("groupBy") ) {
+                foam.mlang.sink.GroupBy groupBy = (foam.mlang.sink.GroupBy) dao.select(MLang.GROUP_BY(pInfo, new foam.mlang.sink.Count()));
+
+                for ( Object key : groupBy.getGroups().keySet() ) {
+                  foam.mlang.sink.Count count = ((foam.mlang.sink.Count) groupBy.getGroups().get(key));
+                  out.println(key + " : " + count.getValue());
+                }
+              } else if ( objPs.toString().startsWith("min") || objPs.toString().startsWith("max") || objPs.toString().startsWith("avg") ) {
```

Comment:
why are you doing this check a second time? it was already done above and can be skipped here by just using an 'else' statement.